src/algebra/continued_fractions/computation/approximations.lean

```diff
@@ -0,0 +1,278 @@
+for the values involved in the continued fractions computation `gcf.of`.
+
+## Main Theorems
+
+- `gcf.of_part_num_eq_one`: shows that all partial numerators `aᵢ` are equal to one.
+- `gcf.of_part_denom_corr_int`: shows that all partial denominators `bᵢ` correspond to an integer.
+- `gcf.of_one_le_nth_part_denom`: shows that `1 ≤ bᵢ`.
+- `gcf.of_succ_nth_fib_le_nth_denom`: shows that the `n`th denominator `Bₙ` is greater or equal than
+  the `n + 1`th fibonacci number `nat.fib (n + 1)`.
+- `gcf.of_le_succ_nth_denom`: shows that `Bₙ₊₁ ≥ bₙ* Bₙ`, where `bₙ` is the `n`th partial denominator
+  of the continued fraction.
+
+## References
+
+- [*Hardy, GH and Wright, EM and Heath-Brown, Roger and Silverman, Joseph*][hardy2008introduction]
+-/
+
+namespace generalized_continued_fraction
+open generalized_continued_fraction as gcf
+
+variables {K : Type*} {v : K} {n : ℕ} [discrete_linear_ordered_field K] [floor_ring K]
+
+/-
+We begin with some lemmas about the stream of `int_fract_pair`s, which presumably are not
+of great interest for the end user.
+-/
+namespace int_fract_pair
+
+/-- Shows that the fractional parts of the stream are in `[0,1)`. -/
+lemma nth_stream_fr_nonneg_lt_one {ifp_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n) :
+  0 ≤ ifp_n.fr ∧ ifp_n.fr < 1 :=
+begin
+  cases n,
+  case nat.zero
+  { have : int_fract_pair.of v = ifp_n, by injection nth_stream_eq,
+    simp [fract_lt_one, fract_nonneg, int_fract_pair.of, this.symm] },
+  case nat.succ
+  { rcases (succ_nth_stream_eq_some_iff.elim_left nth_stream_eq) with ⟨_, _, _, if_of_eq_ifp_n⟩,
+    simp [fract_lt_one, fract_nonneg, int_fract_pair.of, if_of_eq_ifp_n.symm] }
+end
+
+/-- Shows that the fractional parts of the stream are nonnegative. -/
+lemma fr_nonneg {ifp_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n) :
+  0 ≤ ifp_n.fr :=
+(nth_stream_fr_nonneg_lt_one nth_stream_eq).left
+
+/-- Shows that the fractional parts of the stream smaller than one. -/
+lemma fr_lt_one {ifp_n : int_fract_pair K}
+  (nth_stream_eq : int_fract_pair.stream v n = some ifp_n) :
+  ifp_n.fr < 1 :=
+(nth_stream_fr_nonneg_lt_one nth_stream_eq).right
+
+/-- Shows that the integer parts of the stream are at least one. -/
+lemma one_le_succ_nth_stream_b {ifp_succ_n : int_fract_pair K}
+(succ_nth_stream_eq : int_fract_pair.stream v (n + 1) = some ifp_succ_n) :
+  1 ≤ ifp_succ_n.b :=
+begin
+  obtain ⟨ifp_n, nth_stream_eq, stream_nth_fr_ne_zero, ⟨refl⟩⟩ :
+    ∃ ifp_n, int_fract_pair.stream v n = some ifp_n ∧ ifp_n.fr ≠ 0
+    ∧ int_fract_pair.of ifp_n.fr⁻¹ = ifp_succ_n, from
+      succ_nth_stream_eq_some_iff.elim_left succ_nth_stream_eq,
+  change 1 ≤ ⌊ifp_n.fr⁻¹⌋,
+  suffices : 1 ≤ ifp_n.fr⁻¹, by { rw_mod_cast [le_floor], assumption },
+  have : 0 ≤ ifp_n.fr ∧ ifp_n.fr < 1, from nth_stream_fr_nonneg_lt_one nth_stream_eq,
+  cases this with ifp_n_fract_nonneg ifp_n_fract_lt_one,
```

Comment:
I would remove these two lines, and use `fr_nonneg` when necessary.

Reply:
Changed it